src/pytds/tls.py

```diff
@@ -134,7 +134,7 @@ def validate_host(cert, name: bytes) -> bool:
         ext = cert.get_extension(i)
         if ext.get_short_name() == b"subjectAltName":
             s = str(ext)
-            if is_san_matching(s, s_name):
+            if is_san_matching(s.upper(), s_name.upper()):
```

Comment:
can you move logic that does case insensitive comparison inside is_san_matching and add unit test for is_san_matching for such case?